[BUG] query sys.object_dependencies failed error

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. CREATE catalog xxx which is unavailable
2. `select * from sys.object_dependencies`

### Expected behavior (Required)
We can query normally



### Real behavior (Required)
In my scenario, I originally had an iceberg catalog, but I can’t connect to it now.

```log
java.lang.IllegalArgumentException: java.net.UnknownHostException: cluster
        at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:475) ~[hadoop-common-3.3.6.jar:?]
        at org.apache.hadoop.hdfs.NameNodeProxiesClient.createProxyWithClientProtocol(NameNodeProxiesClient.java:134) ~[hadoop-hdfs-client-3.3.6.jar:?]
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:374) ~[hadoop-hdfs-client-3.3.6.jar:?]
        at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:308) ~[hadoop-hdfs-client-3.3.6.jar:?]
        at org.apache.hadoop.hdfs.DistributedFileSystem.initDFSClient(DistributedFileSystem.java:204) ~[hadoop-hdfs-client-3.3.6.jar:?]
        at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:189) ~[hadoop-hdfs-client-3.3.6.jar:?]
        at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3746) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:199) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3857) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3805) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.FileSystem.lambda$get$0(FileSystem.java:604) ~[starrocks-hadoop-ext.jar:?]
        at com.starrocks.connector.hadoop.HadoopExt.doAs(HadoopExt.java:61) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:598) ~[starrocks-hadoop-ext.jar:?]
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365) ~[hadoop-common-3.3.6.jar:?]
        at org.apache.iceberg.hadoop.Util.getFs(Util.java:56) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.hadoop.HadoopInputFile.fromLocation(HadoopInputFile.java:56) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.hadoop.HadoopFileIO.newInputFile(HadoopFileIO.java:90) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.io.ResolvingFileIO.newInputFile(ResolvingFileIO.java:61) ~[iceberg-core-1.2.1.jar:?]
        at com.starrocks.connector.iceberg.io.IcebergCachingFileIO.newInputFile(IcebergCachingFileIO.java:133) ~[starrocks-fe.jar:?]
        at org.apache.iceberg.TableMetadataParser.read(TableMetadataParser.java:266) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.lambda$refreshFromMetadataLocation$0(BaseMetastoreTableOperations.java:189) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.lambda$refreshFromMetadataLocation$1(BaseMetastoreTableOperations.java:208) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:219) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:203) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:196) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.refreshFromMetadataLocation(BaseMetastoreTableOperations.java:208) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.refreshFromMetadataLocation(BaseMetastoreTableOperations.java:185) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.refreshFromMetadataLocation(BaseMetastoreTableOperations.java:180) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.hive.HiveTableOperations.doRefresh(HiveTableOperations.java:176) ~[iceberg-hive-metastore-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.refresh(BaseMetastoreTableOperations.java:97) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreTableOperations.current(BaseMetastoreTableOperations.java:80) ~[iceberg-core-1.2.1.jar:?]
        at org.apache.iceberg.BaseMetastoreCatalog.loadTable(BaseMetastoreCatalog.java:47) ~[iceberg-core-1.2.1.jar:?]
        at com.starrocks.connector.iceberg.hive.IcebergHiveCatalog.getTable(IcebergHiveCatalog.java:98) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.CachingIcebergCatalog.getTable(CachingIcebergCatalog.java:96) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.iceberg.IcebergMetadata.getTable(IcebergMetadata.java:227) ~[starrocks-fe.jar:?]
        at com.starrocks.connector.CatalogConnectorMetadata.getTable(CatalogConnectorMetadata.java:111) ~[starrocks-fe.jar:?]
        at com.starrocks.server.MetadataMgr.lambda$getTable$3(MetadataMgr.java:290) ~[starrocks-fe.jar:?]
        at java.util.Optional.map(Optional.java:265) ~[?:?]
        at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:290) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.BaseTableInfo.getTableByName(BaseTableInfo.java:190) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.BaseTableInfo.getTable(BaseTableInfo.java:154) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.system.sys.SysObjectDependencies.listObjectDependencies(SysObjectDependencies.java:99) ~[starrocks-fe.jar:?]
        at com.starrocks.service.FrontendServiceImpl.listObjectDependencies(FrontendServiceImpl.java:642) ~[starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$listObjectDependencies.getResult(FrontendService.java:4701) ~[starrocks-fe.jar:?]
        at com.starrocks.thrift.FrontendService$Processor$listObjectDependencies.getResult(FrontendService.java:4681) ~[starrocks-fe.jar:?]
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38) ~[libthrift-0.13.0.jar:0.13.0]
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:38) ~[libthrift-0.13.0.jar:0.13.0]
        at com.starrocks.common.SRTThreadPoolServer$WorkerProcess.run(SRTThreadPoolServer.java:311) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.net.UnknownHostException: cluster
```

### StarRocks version (Required)
 - 3.2.1-79ee91d

--- a/fe/fe-core/src/main/java/com/starrocks/catalog/system/sys/SysObjectDependencies.java
+++ b/fe/fe-core/src/main/java/com/starrocks/catalog/system/sys/SysObjectDependencies.java
@@ -32,6 +32,8 @@ import com.starrocks.thrift.TObjectDependencyReq;
 import com.starrocks.thrift.TObjectDependencyRes;
 import com.starrocks.thrift.TSchemaTableType;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.util.Collection;
 import java.util.Optional;
@@ -39,6 +41,9 @@ import java.util.Optional;
 public class SysObjectDependencies {
 
     public static final String NAME = "object_dependencies";
+
+    private static final Logger LOG = LogManager.getLogger(SysObjectDependencies.class);
+
 
     public static SystemTable create() {
         return new SystemTable(SystemId.OBJECT_DEPENDENCIES, NAME, Table.TableType.SCHEMA,
@@ -75,6 +80,10 @@ public class SysObjectDependencies {
             String catalog = Optional.ofNullable(db.getCatalogName())
                     .orElse(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME);
             for (Table table : db.getTables()) {
+                // If it is not a materialized view, we do not need to verify permissions
+                if (!table.isMaterializedView()) {
+                    continue;
+                }
                 // Only show tables with privilege
                 try {
                     Authorizer.checkAnyActionOnTableLikeObject(currentUser, null, db.getFullName(), table);
@@ -82,31 +91,47 @@ public class SysObjectDependencies {
                     continue;
                 }
 
-                if (table.isMaterializedView()) {
-                    MaterializedView mv = (MaterializedView) table;
-                    for (BaseTableInfo refObj : CollectionUtils.emptyIfNull(mv.getBaseTableInfos())) {
-                        TObjectDependencyItem item = new TObjectDependencyItem();
-                        item.setObject_id(mv.getId());
-                        item.setObject_name(mv.getName());
-                        item.setDatabase(db.getFullName());
-                        item.setCatalog(catalog);
-                        item.setObject_type(mv.getType().toString());
+                MaterializedView mv = (MaterializedView) table;
+                for (BaseTableInfo refObj : CollectionUtils.emptyIfNull(mv.getBaseTableInfos())) {
+                    TObjectDependencyItem item = new TObjectDependencyItem();
+                    item.setObject_id(mv.getId());
+                    item.setObject_name(mv.getName());
+                    item.setDatabase(db.getFullName());
+                    item.setCatalog(catalog);
+                    item.setObject_type(mv.getType().toString());
 
-                        item.setRef_object_id(refObj.getTableId());
-                        item.setRef_object_name(refObj.getTableName());
-                        item.setRef_database(refObj.getDbName());
-                        item.setRef_catalog(refObj.getCatalogName());
-                        item.setRef_object_type(Optional.ofNullable(refObj.getTable())
-                                .map(x -> x.getType().toString())
-                                .orElse("UNKNOWN"));
+                    item.setRef_object_id(refObj.getTableId());
+                    item.setRef_object_name(refObj.getTableName());
+                    item.setRef_database(refObj.getDbName());
+                    item.setRef_catalog(refObj.getCatalogName());
+                    item.setRef_object_type(getRefObjectType(refObj, mv.getName()));
 
-                        response.addToItems(item);
-                    }
+                    response.addToItems(item);
                 }
             }
         }
 
         return response;
+    }
+
+    /**
+     * We may not be able to obtain the base table information when external catalog is unavailable
+     *
+     * @param refObj Base tables for materialized views
+     * @param mvName materialized view name
+     * @return base table type
+     */
+    private static String getRefObjectType(BaseTableInfo refObj, String mvName) {
+        String refObjType = "UNKNOWN";
+        try {
+            refObjType = Optional.ofNullable(refObj.getTable())
+                    .map(x -> x.getType().toString())
+                    .orElse("UNKNOWN");
+        } catch (Exception e) {
+            LOG.error("can not get table type error, mv name : {}, error-msg : {}",
+                    mvName, e.getMessage(), e);
+        }
+        return refObjType;
     }
 
 }

--- a/fe/fe-core/src/test/java/com/starrocks/catalog/system/sys/SysObjectDependenciesTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/catalog/system/sys/SysObjectDependenciesTest.java
@@ -1,0 +1,141 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.catalog.system.sys;
+
+import com.starrocks.common.FeConstants;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.DDLStmtExecutor;
+import com.starrocks.sql.plan.ConnectorPlanTestBase;
+import com.starrocks.thrift.TAuthInfo;
+import com.starrocks.thrift.TObjectDependencyReq;
+import com.starrocks.thrift.TObjectDependencyRes;
+import com.starrocks.thrift.TUserIdentity;
+import com.starrocks.utframe.StarRocksAssert;
+import com.starrocks.utframe.UtFrameUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SysObjectDependenciesTest {
+
+    private StarRocksAssert starRocksAssert;
+
+    private ConnectContext connectContext;
+
+    @ClassRule
+    public static TemporaryFolder temp = new TemporaryFolder();
+
+
+    @Before
+    public void setUp() throws Exception {
+        FeConstants.runningUnitTest = true;
+        UtFrameUtils.createMinStarRocksCluster();
+
+        connectContext = UtFrameUtils.createDefaultCtx();
+        starRocksAssert = new StarRocksAssert(connectContext);
+        String createUserSql = "create user if not exists test_mv";
+        DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(createUserSql, connectContext), connectContext);
+    }
+
+    @Test
+    public void testObjectDependencies() throws Exception {
+        starRocksAssert.withDatabase("test")
+                .useDatabase("test")
+                .withTable("CREATE TABLE test.test_mv_base_table\n" +
+                        "(\n" + "    k1 date,\n" + "    k2 int,\n" + "    v1 int sum\n" + ")\n"
+                        + "PARTITION BY RANGE(k1)\n" +
+                        "(\n" + "    PARTITION p1 values [('2022-02-01'),('2022-02-16')),\n"
+                        + "    PARTITION p2 values [('2022-02-16'),('2022-03-01'))\n" + ")\n"
+                        + "DISTRIBUTED BY HASH(k2) BUCKETS 3\n" + "PROPERTIES('replication_num' = '1');")
+                .withMaterializedView("create materialized view test.mv_1\n"
+                        + "PARTITION BY k1\n" + "distributed by hash(k2) buckets 3\n"
+                        + "refresh async\n"
+                        + "as select k1, k2, sum(v1) as total from test_mv_base_table group by k1, k2;")
+                .withMaterializedView("create materialized view test.mv_2\n"
+                        + "PARTITION BY date_trunc('month', k1)\n"
+                        + "distributed by hash(k2) buckets 3\n"
+                        + "refresh async\n"
+                        + "as select k1, k2, sum(v1) as total from test_mv_base_table group by k1, k2;");
+
+
+        String grantSql2 = "GRANT ALL ON TABLE test.test_mv_base_table TO USER `test_mv`@`%`;";
+        String grantSql = "GRANT ALL ON MATERIALIZED VIEW test.mv_1 TO USER `test_mv`@`%`;";
+        String grantSql1 = "GRANT ALL ON MATERIALIZED VIEW test.mv_2 TO USER `test_mv`@`%`;";
+        DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(grantSql2, connectContext), connectContext);
+        DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(grantSql, connectContext), connectContext);
+        DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(grantSql1, connectContext), connectContext);
+
+        TObjectDependencyReq dependencyReq = buildRequest();
+
+        TObjectDependencyRes objectDependencyRes = SysObjectDependencies.listObjectDependencies(dependencyReq);
+        Assert.assertNotNull(objectDependencyRes);
+        Assert.assertEquals(2, objectDependencyRes.getItemsSize());
+        Assert.assertEquals("OLAP", objectDependencyRes.getItems().get(0).getRef_object_type());
+    }
+
+
+    @Test
+    public void testUnknownCatalogObjectDependencies() throws Exception {
+        ConnectorPlanTestBase.mockCatalog(connectContext, temp.newFolder().toURI().toString());
+        starRocksAssert = new StarRocksAssert(connectContext);
+
+
+        String mvName = "test.iceberg_mv";
+
+        starRocksAssert.withDatabase("test")
+                .useDatabase("test")
+                .withMaterializedView("create materialized view " + mvName + " " +
+                        "partition by str2date(d,'%Y-%m-%d') " +
+                        "distributed by hash(a) " +
+                        "REFRESH DEFERRED MANUAL " +
+                        "PROPERTIES (\n" +
+                        "'replication_num' = '1'" +
+                        ") " +
+                        "as select a, b, d, bitmap_union(to_bitmap(t1.c))" +
+                        " from iceberg0.partitioned_db.part_tbl1 as t1 " +
+                        " group by a, b, d;");
+
+        String grantSql1 = "GRANT ALL ON MATERIALIZED VIEW test.iceberg_mv TO USER `test_mv`@`%`;";
+        DDLStmtExecutor.execute(UtFrameUtils.parseStmtWithNewParser(grantSql1, connectContext), connectContext);
+
+
+        TObjectDependencyReq dependencyReq = buildRequest();
+
+
+        TObjectDependencyRes objectDependencyRes = SysObjectDependencies.listObjectDependencies(dependencyReq);
+
+        Assert.assertEquals("ICEBERG", objectDependencyRes.getItems().get(0).getRef_object_type());
+    }
+
+
+    private static TObjectDependencyReq buildRequest() {
+        TObjectDependencyReq dependencyReq = new TObjectDependencyReq();
+        TAuthInfo tAuthInfo = new TAuthInfo();
+
+        TUserIdentity userIdentity = new TUserIdentity();
+        userIdentity.setUsername("test_mv");
+        userIdentity.setHost("%");
+        userIdentity.setIs_domain(false);
+        tAuthInfo.setCurrent_user_ident(userIdentity);
+        tAuthInfo.setUser("root");
+        tAuthInfo.setUser_ip("127.0.0.1");
+        dependencyReq.setAuth_info(tAuthInfo);
+        return dependencyReq;
+    }
+
+
+}

